### DISCUSSION TITLE
1006: Connection dropped by remote peer.

### DISCUSSION THREAD

**Discussion by salbert96** [deepgram/community#897] [General help] [answered]
I am using Speech to Text with streaming (Nova-2). Sometimes the websocket gets closed with this:

```
code: 1006
reason: Connection dropped by remote peer.
```

Any idea, when it can happen? I am using `"@deepgram/sdk": "^3.2.0",`

Here is how I use it:

```javascript
    const deepgram = createClient(apiKey);

    const options: LiveSchema = {
      model: 'nova-2',
      language: 'en-US',

      smart_format: true,

      encoding: 'linear16',
      sample_rate: 48000,

      interim_results: true,
      utterance_end_ms: 1200,
      vad_events: true,
      filler_words: true,

      endpointing: 300,
    };

    const connection = deepgram.listen.live(options);
```

Thank you in advance.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment]
@salbert96  The code 1006 and the reason "Connection dropped by remote peer" typically indicate that a WebSocket connection was abruptly closed by the server or client. This might occur due to network issues, server overload, or deliberate disconnection by the remote peer. 

I noticed you are using endpointing and interim results, We have a pretty [detailed guide](https://developers.deepgram.com/docs/understand-endpointing-interim-results/) on using these features together which may give you some guidance to help resolve this.

**Reply by salbert96** [reply]
Hi @jpvajda , thank you for the answer and the documentation link. 

> was abruptly closed by the server or client. This might occur due to network issues, server overload, or deliberate disconnection by the remote peer

Is there a way to detect when it's closed by a remote peer or server and try to reduce it? I believe we are not closing it, so my assumption is that it's happening on the server side, but I don't know how we can reduce it, or if we should make it in a way that it can recover from disconnection.

**Reply by davidvonthenen** [reply]
Hi @salbert96 

Those connection terminations are usually the result of:
- network interruption occurring anywhere between the client/client-provider to server/server-provider.
- you getting kicked off a node server side because of something like a deployment

You should be able to get a notification perhaps from the SDK about the connection closing. Otherwise, depending on the websocket libs uses get a notification from the WS itself. Worst case, in your keep alive loop, check for a send failure with an error of a "not connected socket".

This looks like JS, so Luke might be able to provide more color.

**Reply by salbert96** [reply]
Hi @dvonthenen thank you for the answer.

> you getting kicked off a node server side because of something like a deployment

For our server deployments, we terminate only if all the connections are closed, I will double-check it though. Can it happen on Deepgram side, when terminating the servers/containers? 

> You should be able to get a notification perhaps from the SDK about the connection closing.

I am receiving the connection close notification with the 1006 code. It happens with around 10 out of 200 connections, so I am trying to understand if there is something we can do.

**Reply by davidvonthenen** [reply]
I was talking about getting kicked off a node on the DG server (an example would be like the deploy of a new model).

A worst case scenario, would be establishing your websocket connection just before a server deploy. The back end tries to do its best to bleed off connections to nodes that need to be updated, but at some point if the connection's a long-running connection it will get booted in favor of the upgrade.

Again, just picking a random example for a connection termination. Not saying it is that at all.

**Reply by davidvonthenen** [reply]
What region are you establishing these connections in?

**Reply by salbert96** [reply, marked as answer]
They are being established from `us-east-1`, we keep the websocket connections with DG for no longer than 10 minutes. 

But I guess if it's expected to happen during deployments, then we will need to have a reconnection logic on our side to recover.

Thank you for all the context. I will revisit, to make sure there are no connection issues on our side. If it starts to be critical, I will ping in this thread ))
